netmiko/garderos/garderos_ssh.py

```diff
@@ -0,0 +1,488 @@
+        Checks whether single config commands executed successfully.
+
+        Automatically exits/enters configuration mode.
+
+        :param config_commands: Multiple configuration commands to be sent to the device
+
+        :param exit_config_mode: Determines whether or not to exit config mode after complete
+
+        :param delay_factor: Deprecated in Netmiko 4.x. Will be eliminated in Netmiko 5.
+
+        :param max_loops: Deprecated in Netmiko 4.x. Will be eliminated in Netmiko 5.
+
+        :param strip_prompt: Determines whether or not to strip the prompt
+
+        :param strip_command: Determines whether or not to strip the command
+
+        :param read_timeout: Absolute timer to send to read_channel_timing. Also adjusts
+        read_timeout in read_until_pattern calls.
+
+        :param config_mode_command: The command to enter into config mode
+
+        :param cmd_verify: Whether or not to verify command echo for each command in config_set
+
+        :param enter_config_mode: Do you enter config mode before sending config commands
+
+        :param error_pattern: Regular expression pattern to detect config errors in the
+        output.
+
+        :param terminator: Regular expression pattern to use as an alternate terminator in certain
+        situations.
+
+        :param bypass_commands: Regular expression pattern indicating configuration commands
+        where cmd_verify is automatically disabled.
+        """
+        # The result of all commands will be collected to config_results
+        config_results = ""
+        # Set delay_factor to given value
+        if delay_factor is None:
+            delay_factor = self.select_delay_factor(0)
+        else:
+            delay_factor = self.select_delay_factor(delay_factor)
+        # Verify if config_commands is an array
+        if config_commands is None:
+            return config_results
+        elif isinstance(config_commands, str):
+            config_commands = (config_commands,)
+        if not hasattr(config_commands, "__iter__"):
+            raise ValueError("Invalid argument passed into send_config_set")
+        # Go to config mode. Use given config_mode_command if necessary.
+        if enter_config_mode:
+            if config_mode_command:
+                config_results += self.config_mode(config_mode_command)
+            else:
+                config_results += self.config_mode()
+        # Send all commands to the router and verify their successful execution
+        for command in config_commands:
+            # Verification is done in send_config_command() function
+            # Will raise error on execution failure
+            result = self.send_config_command(command)
+            config_results = f"{command}\n{result}\n"
+        # Exit config mode if needed
+        if exit_config_mode:
+            self.exit_config_mode()
+        # Return all results
+        # Will only be executed if no error occured
+        return config_results
```

Comment:
What issue do you run into with the standard `send_config_set()` method in Netmiko? 

Why did you need to override it? I say this as it is not maintainable to have a custom-config method for each driver so we should really do everything in our power to use the parent method?

Is it needed for this?

```
        # Verify if configuration command executed successfully
        if command_result != "Set.":
            raise ConfigInvalidException(
                'Error executing configuration command "{}". Device said: {}'.format(
                    command_string, command_result
                )
            )
 ```

Reply:
Garderos basically confirms every correct configuration command by the response "Set.". Otherwise, it will respond with a descriptive error message.
```
grs(config)# helloworld
Unknown.
grs(config)# hello=world
Attribute 'hello' not found in node ''.
grs(config)# system.hello=world
Attribute 'hello' not found in node 'system'.
grs(config)# system.name=???
Attribute 'system.name' incorrect: Incorrect format.
grs(config)# system.name=helloworld
Set.
```
In my understanding, the official implementation of send_config_set() only offers the possibility to blacklist certain error-related string patterns (using error_pattern), correct? However, it appears impossible to reverse the logic by only whitelisting the response "Set." and treating all other responses as errors.

Technically, in regex this would be possible with negative lookahead. Yet, this would only work if regex is matched against the mere respose of the configuration command. Unfortunately, send_config_set() matches error_pattern against the entire config session output (including responses to previously executed configuration commands), so this appears no solution to me. Please correct me if I'm mistaken.

My intention was to implement a "whitelist" version of send_config_set() since a blacklisting approach might not be exhaustive on Garderos GRS.

After all, I agree that code needs to be maintainable. So my suggestion would be that I stick to the error_pattern approach, only overriding the default regex value. I'll rework it and create a new PR.

Reply:
Okay, I think we probably should put a whitelist version of `error_pattern` where it would `error_pattern`. In other words, make a version of `error_pattern` that is a positive match. If the positive match is not encountered then, raise an exception.

Reply:
Revisiting this, would it work to make `error_pattern=r"(Attribute|Unknown)"` (or are there more possible outputs).

Alternatively, we could switch the `error_pattern` search to only search on new data and not on all of the data. So that way we could do the negative lookahead on 'Set.'.